### ISSUE TITLE
[Accessibility][DatePicker] VoiceOver announces "blank" when the focus lands on the dates

### ISSUE THREAD

**Issue by pasangsherpa** [OfficeDev/office-ui-fabric-react#11704]
### Environment Information

- **Package version(s)**: 6.157.0
- **Browser and OS versions**: all

### Describe the issue:
When navigating the dates on the calendar, VoiceOver announces the selected date information followed with "blank". Eg: January 14, 2020, blank

### Please provide a reproduction of the issue in a codepen:
1. Enable VoiceOver
2. Go to https://fabricweb.z5.web.core.windows.net/oufr/7.74.2/#/examples/datepicker
3. Navigate the dates
4. Notice the VoiceOver announces "blank"

<img width="1253" alt="DatePicker Blank" src="https://user-images.githubusercontent.com/2449385/72379567-8fac1500-36c8-11ea-9c78-ec020f986eb8.png">

#### Actual behavior:
VoiceOver announces "October 5 2019 blank" when the focus lands on the dates

#### Expected behavior:
VoiceOver should announce "October 5 2019 selected" when the focus lands on the dates

**Comment by xugao** [comment]
@pasangsherpa  - I can confirm the repro. `button` is set with `role=gridcell`, which seems incorrect and causing the issue. will fix soon
